chcemvediet/apps/anonymization/management/commands/attachment_anonymization.py

```diff
@@ -0,0 +1,82 @@
+import sys
+from optparse import make_option
+
+import magic
+from django.core.files.base import ContentFile
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+
+from poleno.attachments.models import Attachment
+from poleno.utils.misc import squeeze
+from chcemvediet.apps.anonymization.models import AttachmentFinalization
+from chcemvediet.apps.inforequests.models import Action
+
+
+class Command(BaseCommand):
+    args = u'attachment_id [file]'
+    help = squeeze(u"""
+            Creates anonymization for the specified Attachment. The anonymized content is read from
+            the given file or from stdin if no file is specified. If no file is specified and stdin
+            is empty, the command will fail.
+            """)
+
+    option_list = BaseCommand.option_list + (
+        make_option(u'--content_type',
+                    help=squeeze(u"""
+                            Content type of file, e.g. "application/pdf". Automatically guessed from
+                            the file content if not specified.
+                            """)
+                    ),
+        make_option(u'--debug',
+                    default=u'',
+                    help=u'Debug message to the newly created anonymization. Empty by default.'
+                    ),
+        make_option(u'--force',
+                    action=u'store_true',
+                    help=u'Overwrite any existing anonymization for the attachment.'
+                    ),
+    )
+
+    @transaction.atomic
+    def handle(self, *args, **options):
+        if not args:
+            raise CommandError(u'attachment_anonymization takes at least 1 argument (0 given).')
+        elif len(args) > 2:
+            raise CommandError(
+                u'attachment_anonymization takes at most 2 arguments ({} given).'.format(len(args))
+            )
+
+        attachment_pk = args[0]
+        try:
+            attachment = Attachment.objects.attached_to(Action).get(pk=attachment_pk)
+        except (Attachment.DoesNotExist, ValueError):
+            raise CommandError(
+                u'Attachment instance with pk "{}" does not exist.'.format(attachment_pk)
+            )
+        attachments_finalization = (AttachmentFinalization.objects
+                                    .filter(attachment=attachment)
+                                    .successful())
+        if not options[u'force'] and attachments_finalization:
+            raise CommandError(u'Anonymization already exists. Use the --force to overwrite it.')
+
+        if len(args) == 2:
+            filename = args[1]
+            try:
+                with open(filename, u'rb') as file:
+                    content = file.read()
+            except IOError as e:
+                raise CommandError(u'Could not open file: {}.'.format(e))
+        else:
+            content = sys.stdin.read()
+            if not content:
```

Comment:
@martinmacko47 Kontrolujeme sice, ci je `stdin` prazdny, ale nekontrolujeme ci je aj `file` prazdny. Teda moze sa stat, ze ako anonymizacia bude pouzity prazdny file. Osetrime aj tuto situaciu alebo, alebo to nechame tak, ci nastane nejaky pripad, kedy zamerne budeme potrebovat prazdnu anonymizaciu?

Reply:
@viliambalaz IMHO ked admin explicitne uvedie prazdny subor, tak to tak asi chce. Tym prazdnym stdin islo skor o situacie, ked sa mu nepodari spravne spravit pipu a na stdin nebude nic.